config/samples/metallb.io_v1alpha1_peer.yaml

```diff
@@ -0,0 +1,16 @@
+
+apiVersion: metallb.io/v1alpha1
+kind: Peer
+metadata:
+  name: peer-sample1
+  namespace: metallb-system
+spec:
```

Comment:
Similar to our discussion on style for address pools, I think these should be `peerAddress`, `peerASN`, ... to follow Kubernetes API conventions.

Can you add `holdTime`, `peerPort`, and `password` to the sample?  It would just be helpful to see the full set of possible options.  You could have two sample peers: one with minimal required config, and another that shows the full set of options that can be set.

You'll need to add a new `sourceAddress` field.  That's a pretty new option that was just added.

Reply:
@russellb what do u think about something like this, more closer to how BGP is really configured on routers ?
```
apiVersion: metallb.io/v1alpha1
kind: BGPGlobal
metadata:
  name: bgpglobal-sample1
  namespace: metallb-system
spec:
  as: 64500
  routerID: 10.10.10.10
  holdTimer: 20
----
apiVersion: metallb.io/v1alpha1
kind: BGPNeighbor
metadata:
  name: bgpneighbor1
  namespace: metallb-system
spec:
  neighbor: 10.0.0.1
  remoteAs: 64501
  remotePort: 1234
  password: "1234"
  updateSource: "1.2.3.4"
  node-selectors:
  - match-expressions:
    - key: kubernetes.io/hostname
      operator: In
      values: [hostA, hostB]
---
apiVersion: metallb.io/v1alpha1
kind: BGPNeighbor
metadata:
  name: bgpneighbor2
  namespace: metallb-system
spec:
  neighbor: 11.0.0.1
  remoteAs: 64502
  remotePort: 1234
  password: "1234"
  updateSource: "5.6.7.8"

```

Reply:
> @russellb what do u think about something like this, more closer to how BGP is really configured on routers, this is a full BGP configuration for Metallb ?
> 
> ```
> apiVersion: metallb.io/v1alpha1
> kind: AddressPool
> metadata:
>   name: addresspool-bgp
>   namespace: metallb-system
> spec:
>   protocol: bgp
>   addresses:
>     - 172.18.0.100-172.18.0.255
> ---
> apiVersion: metallb.io/v1alpha1
> kind: BGPGlobal
> metadata:
>   name: global
>   namespace: metallb-system
> spec:
>   as: 64500

`ASN:` ?

>   routerID: 10.10.10.10
>   holdTimer: 20

Is it possible that FRR would be configured to talk to 2 different BGP peers over two different networks and use different router IDs on each?  Same for as?  I honestly just don't know.  As long as we aren't losing needed flexibility, then this seems OK.

> ----
> apiVersion: metallb.io/v1alpha1
> kind: BGPNeighbor
> metadata:
>   name: neighbor1
>   namespace: metallb-system
> spec:
> # eBGP example
>   neighbor: 10.0.0.1
>   remoteAs: 64501

`remoteASN` ?

I think acronyms like this should all be capitalized.  Also, what's the thinking with dropping the `n`, or is it a typo?

>   port: 1234
>   passWord: "1234"

just `password` is fine here - 1 word

>   updateSource: "1.2.3.4"
>   nodeSelectors:
>   - matchExpressions:
>     - key: kubernetes.io/hostname
>       operator: In
>       values: [hostA, hostB]

Reply:
I'm also feeling torn on `BGPPeer` vs `BGPNeighbor`.  It sounds like `Neighbor` is more consistent with other networking products, but `BGPPeer` would be more familiar in the context of MetalLB and relatable to other documentation about MetalLB.

I lean toward `BGPPeer`, since I imagine it doesn't matter THAT much ... but I could be convinced otherwise. I also asked some other devs what they think and we can see where that goes.

Reply:
about multiple BGP speaker the idea was to create another BGP global and add the new peers its just like a complete new BGP daemon , but I decided to keep it simple as just rename the fields as you suggested originally